### PR TITLE
hint for unknown variable containing dash

### LINKED ISSUE #868: Arithmetic operator parsing inconsistency
Just a very minor bug/inconsistency but it may be noteworthy. 

The `-` operator apparently needs a space before it when it is preceded by an identifier while the `+` operator does not. E.g.
```
#{
  let a = 2;
  a = a+1;   // fine
  a = 1-a;   // fine
  a = 2-3;   // fine
  a = a -1;  // finer
  a = a-1;   // error
}
```

### PR DESCRIPTION
**Pull request by Marmare314**
Fixes #868. When a unknown variable contains `-` the error message suggests subtraction. Currently this is just done at the first dash: `a-b-c` will result in `did you mean a - b-c`. It would also be possible to check if any split results in valid variables, but it would be more complicated to implement and would not be that much more helpful.

### CODE REVIEW

**Comment by rpitasky** [comment]
Great work! I personally think the added friendliness from looking for a valid suggestion is worth the slight increase in implementation complexity. Since the number of possibilities to search is exponential (`2^(n-1)`, where n is the number of terms separated by dashes), I propose searching if there are only like less than 5 to 7 terms. This can easily be a separate PR, though, I leave this here for you, myself, or others to implement :)

**Comment by laurmaedje** [comment]
alternatively, we could also always suggest a full split a.k.a `a - b - c`

**Comment by rpitasky** [comment]
> alternatively, we could also always suggest a full split a.k.a `a - b - c`

Yes, but this may not be valid output, no? I think that anything the compiler suggests as a fix should be valid code, the alternative is immensely frustrating in my experience.

**Comment by Marmare314** [comment]
> anything the compiler suggests as a fix should be valid code

That's a good point. Maybe the default message should be `did you mean to use subtraction?` instead.

For implementing this properly maybe another approach might be to iterate over the variables and check if they are a prefix. This would be O(num-dashes * num-vars).

File: src/eval/scope.rs [hunk 1]
```diff
@@ -43,8 +43,8 @@ impl<'a> Scopes<'a> {
             .chain(self.base.map(|base| base.global.scope()))
             .find_map(|scope| scope.get(var))
             .ok_or_else(|| {
-                if let Some((a, b)) = var.split_once('-') {
-                    eco_format!("unknown variable: {} – did you mean {} - {}?", var, a, b)
+                if var.contains('-') {
+                    eco_format!("unknown variable: {} – if you meant to use subtraction, try adding space around the minus sign.", var)
```
Comment (laurmaedje) [inline src/eval/scope.rs:47]:
```suggestion
                    eco_format!("unknown variable: {} – if you meant to use subtraction, try adding spaces around the minus sign.", var)
```

File: tests/typ/compiler/hint.typ [hunk 2]
```diff
@@ -11,6 +11,12 @@
...
     a = a -1
 
-    // Error: 9-12 unknown variable: a-1 – did you mean a - 1?
+    // Error: 9-12 unknown variable: a-1 – if you meant to use subtraction, try adding space around the minus sign.
     a = a-1
 }
+
+---
+#{
+    // Error: 5-8 unknown variable: a-1 – if you meant to use subtraction, try adding space around the minus sign.
```
Comment (laurmaedje) [inline tests/typ/compiler/hint.typ:20]: should use 2 spaces of indent

**Pull request merged**